lms/djangoapps/email_marketing/tasks.py

```diff
@@ -294,6 +300,97 @@ def _get_course_content(course_url, sailthru_client, email_config):
     return response
 
 
+def _get_or_create_user_list_for_site(sailthru_client, site=None, default_list_name=None):
+    """
+    Get the user list name from cache if exists else create one and return the name,
+    callers of this function should perform the enabled check of email config.
+    :param: sailthru_client
+    :param: site
+    :param: default_list_name
+    :return: list name if exists or created else return None
+    """
+    if site and site.id != settings.SITE_ID:
+        list_name = site.domain.replace(".", "_") + "_user_list"
+    else:
+        list_name = default_list_name
+
+    sailthru_list = _get_or_create_user_list(sailthru_client, list_name)
+    return list_name if sailthru_list else default_list_name
+
+
+def _get_or_create_user_list(sailthru_client, list_name):
+    """
+    Get list from sailthru and return if list_name exists else create a new one
+    and return list data for all lists.
+    :param sailthru_client
+    :param list_name
+    :return sailthru list
+    """
+    sailthru_list_cache = cache.get(SAILTHRU_LIST_CACHE_KEY)
+    if not sailthru_list_cache or not sailthru_list_cache.get(list_name):
+        sailthru_list_cache = _get_user_list_from_email_marketing_provider(sailthru_client)
+        if sailthru_list_cache:
+            cache.set(SAILTHRU_LIST_CACHE_KEY, sailthru_list_cache)
+
+    sailthru_list = sailthru_list_cache.get(list_name) if sailthru_list_cache else None
+
+    if not sailthru_list:
+        is_created = _create_user_list(sailthru_client, list_name)
+        if is_created:
+            sailthru_list_cache = _get_user_list_from_email_marketing_provider(sailthru_client)
+            if sailthru_list_cache:
```

Comment:
I think you want to set the cache here even if the call to `_get_user_list_from_email_marketing_provider`, otherwise you would end up with an invalid cache (a new list has been created but it is not in the cache).